aten/src/ATen/native/ComparisonUtils.cpp

```diff
@@ -14,23 +15,45 @@ class Tensor;
 
 namespace native {
 
-template<typename O, typename C>
-void _assert_match(const O& original, const C& compared, const std::string& name) {
-  if (compared) {
-    bool equal = (original == compared.value());
-    if (!equal) {
-      std::stringstream msg;
-      msg << "Tensor " << name << " mismatch!";
-      AT_ASSERT(equal, msg.str());
+void _assert_equal(const std::string& name, const bool& condition) {
+    if (!condition) {
+        std::stringstream msg;
+        msg << "Tensor " << name << " mismatch!";
+        AT_ASSERT(condition, msg.str());
+    }
+}
+
+void _assert_match(const c10::SymIntArrayRef& original, const c10::OptionalArrayRef<long int>& compared, const std::string& name) {
+    if (compared) {
+        auto sym_compared = c10::SymIntArrayRef(
+            reinterpret_cast<const c10::SymInt*>(compared.value().data()), compared.value().size());
+        _assert_equal(name, original == sym_compared);
```

Comment:
@ezyang you said this was sus. I super duper agree. Whats the right way to do this?